src/NuGet.Indexing/Extraction/DocumentCreator.cs

```diff
@@ -261,6 +263,56 @@ private static void AddRequiresLicenseAcceptance(Document document, IDictionary<
             }
         }
 
+        private static void AddPackageTypes(Document document, IDictionary<string, string> package)
+        {
+            string value;
+            StringBuilder searchIndex = new StringBuilder();
+            if (package.TryGetValue("flattenedPackageTypes", out value))
+            {
+                if (!string.IsNullOrWhiteSpace(value))
+                {
+                    AddField(document, "FlattenedPackageTypes", value, Field.Index.NOT_ANALYZED);
+
+                    using (var textWriter = new StringWriter())
+                    {
+                        using (var jsonWriter = new JsonTextWriter(textWriter))
+                        {
+                            jsonWriter.WriteStartArray();
+                            foreach (var packageType in value.Split('|'))
+                            {
+                                string[] fields = packageType.Split(':');
+                                if (fields.Length > 0)
+                                {
+                                    jsonWriter.WriteStartObject();
+                                    jsonWriter.WritePropertyName("Type");
+                                    jsonWriter.WriteValue(fields[0]);
+                                    searchIndex.Append(fields[0]);
+                                    searchIndex.Append(" ");
+                                    if (fields.Length > 1)
+                                    {
+                                        jsonWriter.WritePropertyName("Version");
+                                        jsonWriter.WriteValue(fields[1]);
+                                    }
+
+                                    jsonWriter.WriteEndObject();
+                                }
+                            }
+
+                            jsonWriter.WriteEndArray();
+                            jsonWriter.Flush();
+                            textWriter.Flush();
+
+                            var packageTypes = textWriter.ToString();
```

Comment:
I think you should be able to optimize similar thing here that AddFlattenedPackageTypes did, that we talked offline.

```
var searchIndex = string.Join(" ", value.Split("|").Select(packageType => packageType.Split(':')[0]).ToList());
```

Might need to add a length check in there as well I guess. If that affects readability a lot, I am fine with the way it is.